modules/videoio/src/cap.cpp

```diff
@@ -371,45 +371,56 @@ CV_IMPL CvCapture * cvCreateFileCapture (const char * filename)
 CV_IMPL CvVideoWriter* cvCreateVideoWriter( const char* filename, int fourcc,
                                             double fps, CvSize frameSize, int is_color )
 {
-    // If none of the writers is used
-    // these statements suppress 'unused parameter' warnings.
+    return cvCreateVideoWriterWithPreference(CV_CAP_ANY, filename, fourcc, fps, frameSize, is_color);
+}
+
+CvVideoWriter* cvCreateVideoWriterWithPreference(int apiPreference, const char* filename, int fourcc,
+                                            double fps, CvSize frameSize, int is_color )
+{
     CV_UNUSED(frameSize);
     CV_UNUSED(is_color);
 
-    //CV_FUNCNAME( "cvCreateVideoWriter" );
-
     CvVideoWriter *result = 0;
 
     if(!fourcc || !fps)
         TRY_OPEN(result, cvCreateVideoWriter_Images(filename))
```

Comment:
Maybe we could define additional API IDs for built-in MotionJPEG and Images backends? Or maybe we should add new set of IDs for _VideoWriter_ specifically: _WRI_ANY_, _WRI_FFMPEG_, etc.? What do you think?